core/Makefile

```diff
@@ -75,6 +78,28 @@ else ifeq ($(words $(MAKECMDGOALS)),1)
 	endif
 endif
 
+# We always want these libraries to be dynamically linked even when the
+# user requests a static build.  These needs to be listed first, so even
+# if they're listed again in the static section, they will be dynamically
+# linked.
+ALWAYS_DYN_LIBS := -lpthread -ldl
+# These libraries are not supported by pkg-config.
+ALWAYS_LIBS := -lpcap -ldl
+# If pkg-config is available, we just need a list of the dependecies.
+PKG_CONFIG_DEPS := libglog gflags protobuf grpc++ libunwind zlib
+# If pkg-config is not available, we need to list the libs we depend on.
+NO_PKG_CONFIG_LIBS := -lglog -lgflags -lprotobuf -lgrpc++ -lunwind -lz
+# If pkg-config is not available and we're static linking, we also need
+# the indirect dependecies.  This is annoying, because they may change
+# in future versions.
+NO_PKG_CONFIG_LIBS_INDIRECT := -lgrpc -lssl -lcrypto -llzma
+
+ifeq ($(HAS_PKG_CONFIG), yes)
+    PKG_CFLAGS = $(shell $(PKG_CONFIG) --cflags $(PKG_CONFIG_DEPS))
```

Comment:
Can this be statically defined (with `:=`)?

Reply:
Thanks for the suggestion! I did that here and in the other places I was shelling out to call `pkg-config`